File: csharp-peer/src/models/Player.cs
using Fomoxa;

namespace Models;

[Network]
[Codec("edge")]
public class Player
{
    [Network("u32")]
    [Codec("edge")]
    public uint Id { get; set; }

    [Network("f32")]
    [Codec("edge")]
    public float X { get; set; }

    [Network("f32")]
    [Codec("edge")]
    public float Y { get; set; }
}

File: rust-peer/src/models/player.rs
use fomoxa_attributes::*;

#[network]
#[codec(edge)]
#[derive(Debug, Default, Clone, Copy, PartialEq)]
pub struct Player {
    #[network(u32)]
    #[codec(edge)]
    pub id: u32,

    #[network(f32)]
    #[codec(edge)]
    pub x: f32,

    #[network(f32)]
    #[codec(edge)]
    pub y: f32,
}
